internal/customer/handlers/grpc/order.go

```diff
@@ -1,50 +1,123 @@
-		Meats:     ProductConv(rmr.Menu.Meats),
-		Soups:     ProductConv(rmr.Menu.Soups),
-		Drinks:    ProductConv(rmr.Menu.Drinks),
-		Desserts:  ProductConv(rmr.Menu.Desserts),
-	}
-	return amr, nil
-}
-
-func ProductConv(p []*restaurant.Product) []*pb.Product {
-	var sl []*pb.Product
-	for _, v := range p {
-		a := &pb.Product{
-			Uuid:        v.Uuid,
-			Name:        v.Name,
-			Description: v.Description,
-			Type:        pb.CustomerProductType(v.Type),
-			Weight:      v.Weight,
-			Price:       v.Price,
-			CreatedAt:   v.CreatedAt,
-		}
-		sl = append(sl, a)
-	}
-	return sl
-}
+package handlerscustomer
+
+import (
+	"context"
+	"fmt"
+	"github.com/MikhailMishutkin/FoodOrdering/internal/types"
+	"github.com/MikhailMishutkin/FoodOrdering/proto/pkg/restaurant"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"log"
+	"strconv"
+	"time"
+
+	pb "github.com/MikhailMishutkin/FoodOrdering/proto/pkg/customer"
+)
+
+func (s *CustomerService) CreateOrder(ctx context.Context, in *pb.CreateOrderRequest) (*pb.CreateOrderResponse, error) {
+	log.Println("CreateOrder was invoked")
+	usId, err := strconv.Atoi(in.UserUuid)
+	if err != nil {
+		return nil, fmt.Errorf("Can't conv user id in CreateOrder: %v\n", err)
+	}
+
+	req := &types.OrderRequest{
+		UserUuid:  usId,
+		Salads:    convProductItem(in.Salads),
+		Garnishes: convProductItem(in.Garnishes),
+		Meats:     convProductItem(in.Meats),
+		Soups:     convProductItem(in.Soups),
+		Drinks:    convProductItem(in.Drinks),
+		Desserts:  convProductItem(in.Desserts),
+	}
+
+	err = s.cs.CreateOrder(req)
+
+	return &pb.CreateOrderResponse{}, err
+}
+
+func (s *CustomerService) GetActualMenu(ctx context.Context, in *pb.GetActualMenuRequest) (*pb.GetActualMenuResponse, error) {
+	log.Println("GetActualMenu was invoked")
+
+	t := time.Now()
+	t1 := t.AddDate(0, 0, 1)
+	ts := timestamppb.New(t1)
+
+	fmt.Println(ts, s.client) //смотрим
+
+	request, err := s.client.GetMenu(context.Background(), &restaurant.GetMenuRequest{
+		OnDate: ts,
+	})
+	if err != nil {
+		log.Println("Can't get the menu from restaurant", err)
+		return nil, err
+	}
+
+	result := &pb.GetActualMenuResponse{
+		Salads:    convertProducts(request.Menu.Salads),
+		Garnishes: convertProducts(request.Menu.Garnishes),
+		Meats:     convertProducts(request.Menu.Meats),
+		Soups:     convertProducts(request.Menu.Soups),
+		Drinks:    convertProducts(request.Menu.Drinks),
+		Desserts:  convertProducts(request.Menu.Desserts),
+	}
+
+	return result, err
+}
+
+func convProductItem(slOI []*pb.OrderItem) (slTypOI []*types.OrderItem) {
+
+	for _, v := range slOI {
+		pId, err := strconv.Atoi(v.ProductUuid)
+		if err != nil {
+			fmt.Errorf("Can't conv user id in CreateOrder: %v\n", err)
+		}
+		typOI := &types.OrderItem{
+			Count:       int(v.Count),
+			ProductUuid: pId,
+		}
+		slTypOI = append(slTypOI, typOI)
+	}
+	return slTypOI
+}
+
+func convertProducts(res []*restaurant.Product) []*pb.Product {
+	var resPb []*pb.Product
```

Comment:
лучше заранее выделить под это память так как ты знаешь длину итогового слайза:

```suggestion
	resPb := make([]*pb.Product, 0, len(res))
```